source/PlayerInfo.cpp

```diff
@@ -1704,31 +1710,53 @@ bool PlayerInfo::TakeOff(UI *ui)
 				int64_t cost = depreciation.Value(outfit.first, day, outfit.second);
 				for(int i = 0; i < outfit.second; ++i)
 					stockDepreciation.Buy(outfit.first, day, &depreciation);
-				income += cost;
+				outfitsIncome += cost;
 			}
-		else
+		else if(planet->HasOutfitter())
 			for(const auto &outfit : cargo.Outfits())
 			{
 				// Transfer the outfits from cargo to the storage on this planet.
 				if(!outfit.second)
 					continue;
 				cargo.Transfer(outfit.first, outfit.second, *Storage(true));
-			}
+			 }
+
 	}
-	accounts.AddCredits(income);
+
+	accounts.AddCredits(commoditiesIncome + outfitsIncome);
 	cargo.Clear();
 	stockDepreciation = Depreciation();
-	if(sold)
+
+	// Report how much excess cargo was sold, and what profit you earned.
+	ostringstream out;
+
+	if(commoditiesSold)
 	{
-		// Report how much excess cargo was sold, and what profit you earned.
-		ostringstream out;
-		out << "You sold " << Format::CargoString(sold, "excess cargo") << " for " << Format::CreditString(income);
-		if(totalBasis && totalBasis != income)
-			out << " (for a profit of " << Format::CreditString(income - totalBasis) << ").";
+		// We check against income for systems like Postverta where trading is disabled by having all prices at 0.
+		if(commoditiesIncome)
+		{
+			out << "You sold " << Format::CargoString(commoditiesSold, "excess cargo") << " for " << Format::CreditString(commoditiesIncome);
+			if(totalBasis && totalBasis != commoditiesIncome)
+				out << " (for a profit of " << Format::CreditString(commoditiesIncome - totalBasis) << ").";
+			else
+				out << ".";
+		}
 		else
-			out << ".";
-		Messages::Add(out.str(), Messages::Importance::High);
+			out << "You dumped " << Format::CargoString(commoditiesSold, "excess cargo.");
+		out << " ";
```

Comment:
So I'm unsure what this modification proposes is it this :
```
if(commoditiesSold) {
...
 if(outfitsSold)
        out << " ";
}
```
Which I find redundant since the extra white space at the end of the message doesn't change anything.
Or is it this :
```
 if(outfitsSold)
 {
      out << " "
}
```
Which is not correct as one could sell/store/dump outfits without selling/dumping commodities.

Reply:
The first one, but yeah it won't change much.

Reply:
Seeing #9060 and #9117 this change would actually be beneficial. I'll commit it.